sunpy/map/mapbase.py

```diff
@@ -2739,6 +2739,13 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
         outmap = GenericMap(output_array, target_wcs.to_header(),
                             plot_settings=self.plot_settings)
 
+        # Check rsun mismatch
+        rsun_map = self.meta["rsun_ref"]
+        rsun_target = outmap.meta["rsun_ref"]
+        if rsun_map != rsun_target:
+            warn_user("rsun mismatch detected. "
```

Comment:
Could you print the values and the names of the map that the different `rsun` values are coming from so this error message is clearer? Something like `warn_user(f"rusn mismatch detected: {self.name}.rsun_obs={self.rsun_obs}; {outmap.name}.rsun_obs={outmap.rsun_obs}"). This will make it easier for the user to understand which maps have different values and how different those values are.

Reply:
Sure thats a good idea.